### PR TITLE
feat: add event logs for vault config updates

### PR DESCRIPTION
**Pull request by hrharder**
fixes: #128 

Adds granular events for the following changes to a vaults parameters/configurations:

- ERC20 name set/updated
- ERC20 symbol set/updated
- Guest list set/updated
- Rewards address set/updated
- Deposit limit set/updated
- Performance fee set/updated
- Management fee set/updated
- Guardian address set/updated
- Emergency shutdown state set (not logged in initialization)
- Withdrawal queue set/updated

Are tests necessary for something this simple? Any feedback/requests on naming or indexed parameters or anything like that? First time using vyper as so appreciate any possible improvements.

Also -- to confirm, no linter for vyper? Followed pattern of two newlines after event declarations.

h/t @fubuloubu

### CODE REVIEW

File: contracts/Vault.vy [hunk 1]
```diff
@@ -116,6 +116,46 @@ event StrategyReported:
...
 
 
+event SetName:
+    name: String[64] # New active Vault ERC20 name
+
+
+event SetSymbol:
+    symbol: String[32] # New active Vault ERC20 symbol
+
+
```
Comment (fubuloubu) [inline contracts/Vault.vy:126]: Probably don't need these, should only really be used once or twice

File: contracts/Vault.vy [hunk 2]
```diff
@@ -207,6 +247,13 @@ def __init__(
             convert(self, bytes32)
         )
     )
+    log SetName(self.name)
+    log SetSymbol(self.symbol)
+    log SetRewards(self.rewards)
+    log SetGuardian(self.guardian)
+    log SetPerformanceFee(self.performanceFee)
+    log SetManagementFee(self.managementFee)
+    log SetDepositLimit(self.depositLimit)
```
Comment (fubuloubu) [inline contracts/Vault.vy:256]: Use the input values/literals/constants for these, saving like ~1950 gas each or so

Also, put each one after where it's set

File: contracts/Vault.vy [hunk 3]
```diff
@@ -116,6 +116,46 @@ event StrategyReported:
...
+event SetPerformanceFee:
+    performanceFee: indexed(uint256) # New active performance fee
+
+
+event SetManagementFee:
+    managementFee: indexed(uint256) # New active management fee
+
+
+event SetGuardian:
+    guardian: indexed(address) # Address of the active guardian
```
Comment (fubuloubu) [inline contracts/Vault.vy:148]: No need to index these

Only index `strategy` address in `UpdateStrategy*` events

**Review by fubuloubu — CHANGES_REQUESTED** [review]
Broadly, could use you use `Update*` instead? I know it's a bit more chars, but is more meaningful. Unless you have another suggestion. `Set*` doesn't see right to me.

File: contracts/Vault.vy (hunk 2 above)
Comment (hrharder) [reply contracts/Vault.vy:256]: cool -- didn't know this re. gas

**Comment by hrharder** [comment]
@fubuloubu addressed comments in [a56619d](https://github.com/iearn-finance/yearn-vaults/pull/131/commits/a56619d372c3b6316bfe85cf787466e1b3e61d93) and added events for strategy param updates -- thank you

**Review by banteg — APPROVED** [review]

File: contracts/Vault.vy [hunk 4]
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
...
+    oldVersion: indexed(address) # Old version of the strategy to be upgraded
+    newVersion: indexed(address) # New version of the strategy
+
+
+event StrategyRevoked:
+    strategy: indexed(address) # Address of the strategy that is revoked
+
+
+event StrategyRemoved:
+    strategy: indexed(address) # Address of the strategy that is removed
```
Comment (fubuloubu) [inline contracts/Vault.vy:180]:
```suggestion
event StrategyRemovedFromQueue:
    strategy: indexed(address) # Address of the strategy that is removed from the withdrawal queue
```

Should also have `StrategyAddedToQueue`?

File: contracts/Vault.vy [hunk 5]
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
...
+    strategy: indexed(address) # Address of the strategy for the rate limit adjustment
+    rateLimit: uint256 # The new rate limit for the strategy
+
+
+event UpdateStrategyPerformanceFee:
+    strategy: indexed(address) # Address of the strategy for the performance fee adjustment
+    performanceFee: uint256 # The new performance fee for the strategy
+
+
+event UpdateStrategy:
```
Comment (fubuloubu) [inline contracts/Vault.vy:170]: Change to `StrategyMigrated`

File: contracts/Vault.vy [hunk 6]
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
...
+
+
+event UpdateStrategyRateLimit:
+    strategy: indexed(address) # Address of the strategy for the rate limit adjustment
+    rateLimit: uint256 # The new rate limit for the strategy
+
+
+event UpdateStrategyPerformanceFee:
+    strategy: indexed(address) # Address of the strategy for the performance fee adjustment
+    performanceFee: uint256 # The new performance fee for the strategy
```
Comment (fubuloubu) [inline contracts/Vault.vy:167]: Maybe these could be `StrategyUpdate*` instead? To match the rest

File: contracts/Vault.vy [hunk 7]
```diff
@@ -116,6 +116,70 @@ event StrategyReported:
     debtLimit: uint256
 
 
+event UpdateGovernance:
+    governance: indexed(address) # New active governance
```
Comment (fubuloubu) [inline contracts/Vault.vy:120]: Not sure if this needs to be indexed

**Review by fubuloubu — APPROVED** [review]
A couple nitpicks, but overall looking good!

**Comment by hrharder** [comment]
updated

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
@hrharder good to go? lgtm

**Comment by hrharder** [comment]
@fubuloubu indeed it is -- send it, at your leisure

**Pull request merged**